### PR TITLE
Add opcodes - Js, Jns, Fldpi, Fxch, Fpatan, Fscale

### PR DESCRIPTION
**Pull request by tuday2**
Draft -- need some assistance

### CODE REVIEW

File: MBBSEmu/CPU/CPUCore.cs [hunk 1]
```diff
@@ -3365,6 +3420,63 @@ private void Op_Fld1()
...
+        [MethodImpl(OpcodeCompilerOptimizations)]
+        private void Op_Fxch()
+        {
+            var source = GetOperandValueDouble(_currentInstruction.Op0Kind, EnumOperandType.Source);
+            var ST0 = FpuStack[Registers.Fpu.GetStackTop()];
+            var ST1 = FpuStack[Registers.Fpu.GetStackPointer(Register.ST1)];
+
+            //TODO Handle invalid and infinity cases
+
+            (ST0, ST1) = (ST1, ST0);
```
Comment (paladine) [inline MBBSEmu/CPU/CPUCore.cs:3437]: doesn't this just change the local variable values and not the values actually in the FpuStack?

File: MBBSEmu.Tests/CPU/JS_JNS_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        public void JS_Test(bool signFlagValue, ushort ipValue)
+        {
+            Reset();
+
+            var instructions = new Assembler(16);
+            instructions.js(0);
+            CreateCodeSegment(instructions);
+
+            if (signFlagValue)
+                mbbsEmuCpuRegisters.F = mbbsEmuCpuRegisters.F.SetFlag((ushort)EnumFlags.SF);
```
Comment (enusbaum) [inline MBBSEmu.Tests/CPU/JS_JNS_Tests.cs:22]: You can now use `mbbsEmuCpuRegisters.SignFlag = signFlagValue;`

File: MBBSEmu.Tests/CPU/JS_JNS_Tests.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        public void JNS_Test(bool signFlagValue, ushort ipValue)
+        {
+            Reset();
+
+            var instructions = new Assembler(16);
+            instructions.jns(0);
+            CreateCodeSegment(instructions);
+
+            if (signFlagValue)
+                mbbsEmuCpuRegisters.F = mbbsEmuCpuRegisters.F.SetFlag((ushort)EnumFlags.SF);
```
Comment (enusbaum) [inline MBBSEmu.Tests/CPU/JS_JNS_Tests.cs:57]: See Above

File: MBBSEmu.Tests/CPU/FXCH_Tests.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        public void FXCH_Test(float ST0Value, float ST1Value)
+        {
+            Reset();
+
+            mbbsEmuCpuRegisters.Fpu.SetStackTop(1);
+            mbbsEmuCpuCore.FpuStack[1] = ST0Value; //ST0
+            mbbsEmuCpuCore.FpuStack[0] = ST1Value; //ST1
+
+            var instructions = new Assembler(16);
+            instructions.fxch(st0, st1);
```
Comment (enusbaum) [inline MBBSEmu.Tests/CPU/FXCH_Tests.cs:24]: I would add another test that might switch `ST0` and `ST4` as well

**Review by enusbaum — CHANGES_REQUESTED** [review]
Only the three small changes and it looks good to me!

File: MBBSEmu/CPU/CPUCore.cs [hunk 5]
```diff
@@ -3365,6 +3420,63 @@ private void Op_Fld1()
...
 
+        /// <summary>
+        ///     Floating Point Load Operation (x87)
+        ///
+        ///     Exchanges the contents of registers ST(0) and ST(i). If no source operand is specified, the contents of ST(0) and ST(1) are exchanged.
+        /// </summary>
+        [MethodImpl(OpcodeCompilerOptimizations)]
+        private void Op_Fxch()
+        {
+            var source = GetOperandValueDouble(_currentInstruction.Op0Kind, EnumOperandType.Source);
```
[the remark this answers is not in the document]
Comment (tuday2) [reply MBBSEmu/CPU/CPUCore.cs]: Per our convo source/destination covers this and proved by test

**Review by enusbaum — APPROVED** [review]

**Pull request merged**